src/partest/scala/tools/partest/nest/AbstractRunner.scala

```diff
@@ -73,26 +73,32 @@ class AbstractRunner(val config: RunnerSpec.Config, protected final val testSour
     f"$word $testNumber - $testIdent%-40s$reasonString$durationString"
   }
 
-  def reportTest(state: TestState, info: TestInfo, durationMs: Long, diffOnFail: Boolean, logOnFail: Boolean): Unit = {
-    if (terse && state.isOk) printDot()
-    else {
-      echo(statusLine(state, durationMs))
-      if (!state.isOk) {
-        def showLog() = if (info.logFile.canRead) {
-          echo(bold(cyan(s"##### Log file '${info.logFile}' from failed test #####\n")))
-          echo(info.logFile.fileContents)
-        }
+  def reportTest(state: TestState, info: TestInfo, durationMs: Long, diffOnFail: Boolean, logOnFail: Boolean): List[String] = {
+    def errInfo: List[String] = {
+      def showLog() =
+        if (info.logFile.canRead) List (
+          bold(cyan(s"##### Log file '${info.logFile}' from failed test #####\n")),
+          info.logFile.fileContents
+        ) else Nil
+      val diffed = 
         if (diffOnFail) {
           val differ = bold(red("% ")) + "diff "
-          val diffed = state.transcript find (_ startsWith differ)
-          diffed match {
-            case Some(diff) => echo(diff)
+          state.transcript.find(_ startsWith differ) match {
+            case Some(diff) => diff :: Nil
             case None if !logOnFail && !verbose => showLog()
-            case _ => ()
+            case _ => Nil
           }
-        }
-        if (logOnFail) showLog()
-      }
+        } else Nil
+      val logged = if (logOnFail) showLog() else Nil
+      diffed ::: logged
+    }
+    if (terse) {
+      if (state.isOk) { printDot() ; Nil }
+      else { printEx() ; statusLine(state, durationMs) :: errInfo }
+    } else {
+      echo(statusLine(state, durationMs))
+      if (!state.isOk) errInfo.foreach(echo)
+      Nil
```

Comment:
Is this supposed to return `Nil` when there were errors?

Reply:
IIRC this is just a way to smuggle terse-mode error output to be printed later. If not terse, the output is immediate. I also remember not liking the hack at L313. So this interaction could be encapsulated better.